app/actions/remote/post.ts

```diff
@@ -26,6 +26,84 @@ type AuthorsRequest = {
     error?: unknown;
 }
 
+export const createPost = async (serverUrl: string, post: Partial<Post>, files: FileInfo[] = []): Promise<{data?: boolean; error?: any}> => {
+    const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
+    if (!operator) {
+        return {error: `${serverUrl} database not found`};
+    }
+
+    const currentUserId = queryCurrentUserId(operator.database);
+    const timestamp = Date.now();
+    const pendingPostId = post.pending_post_id || `${currentUserId}:${timestamp}`;
+
+    // Check if is sending this post id
```

Comment:
On V1 we used to check here whether this post id was being sent (https://github.com/mattermost/mattermost-mobile/blob/856d8bd05f02e453d5280fbcff2b39b2f0ff48ab/app/mm-redux/actions/posts.ts#L180). Not sure if needed, or how would be the right way to handle this.

Reply:
yes, this is to avoid sending the same post twice, here you could store the post into the db and have the id match the pendingPostId, then query the DB to determine if the pendingPostId already exists and the `props` of the post does not have a `failed: true` prop